x-pack/plugins/cases/public/components/files/file_preview_modal.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+
+import type { FileJSON } from '@kbn/shared-ux-file-types';
+
+import {
+  EuiModalBody,
+  EuiModalFooter,
+  EuiButton,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+} from '@elastic/eui';
+import { FileImage } from '@kbn/shared-ux-file-image';
+
+import { APP_ID } from '../../../common';
+import { CASES_FILE_KINDS } from '../../files';
+import * as i18n from './translations';
+import { isImage } from './utils';
+
+interface FilePreviewModalProps {
+  closeModal: () => void;
+  getDownloadHref: (args: Pick<FileJSON<unknown>, 'id' | 'fileKind'>) => string;
+  selectedFile: FileJSON;
+}
+
+export const FilePreviewModal = ({
+  closeModal,
+  selectedFile,
+  getDownloadHref,
+}: FilePreviewModalProps) => {
+  return (
+    <EuiModal maxWidth={false} onClose={closeModal}>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>{selectedFile?.name}</EuiModalHeaderTitle>
+      </EuiModalHeader>
+      <EuiModalBody>
+        {isImage(selectedFile) && (
```

Comment:
> I believe we're checking in the column to prevent rendering a modal for a non-image correct?

I don't expect it to happen and that is already taken into account. It is more of a sanity check 😄